src/ac_training_lab/pioreactor/on_reactor.py

```diff
@@ -0,0 +1,509 @@
+    else:
+        print(f"Failed to update stirring RPM. Status code: {response.status_code}")
+
+def set_led_intensity(worker, experiment, brightness_value, led):
+    url = f"http://pioreactor.local/api/workers/{worker}/jobs/run/job_name/led_intensity/experiments/{experiment}"
+    headers = {"Content-Type": "application/json"}
+    payload = {
+        'env': {"EXPERIMENT": experiment, "JOB_SOURCE": "user"},
+        "options": {led: brightness_value, "source_of_event": "UI"}
+        }
+
+    response = requests.patch(url, headers=headers, data=json.dumps(payload))
+    if response.status_code == 202:
+        print(f"LED intensity set to {brightness_value} for worker {worker}.")
+    else:
+        print(f"Failed to set LED intensity. Status code: {response.status_code}")
+
+def get_temperature_readings(client, reactor, experiment, filter_mod, lookback):
+    url = f"http://pioreactor.local/api/experiments/{experiment}/time_series/temperature_readings"
+    params = {"filter_mod_N": filter_mod, "lookback": lookback}
+    
+    response = requests.get(url, params=params)
+    if response.status_code == 200:
+        print(f"Temperature readings retrieved for experiment {experiment}.")
+        print(response.json())
+        temperature_data = response.json()
+        client.publish(f'pioreactor/{reactor}/temperature', json.dumps(temperature_data))
+    else:
+        print(f"Failed to retrieve temperature readings. Status code: {response.status_code}")
+
+def get_experiments(client):
+    url = "http://pioreactor.local/api/experiments"
+    headers = {"Content-Type": "application/json"}
+    
+    response = requests.get(url, headers=headers)
+    if response.status_code == 200:
+        experiments = response.json()
+        client.publish('pioreactor/experiments', json.dumps(experiments))
+    else:
+        print(f"Failed to retrieve experiments. Status code: {response.status_code}")
+
+def get_reactors(client, experiment):
+    url = f"http://pioreactor.local/api/experiments/{experiment}/workers"
+    headers = {"Content-Type": "application/json"}
+    
+    response = requests.get(url, headers=headers)
+    if response.status_code == 200:
+        reactors = response.json()
+        client.publish('pioreactor/reactors', json.dumps(reactors))
+    else:
+        print(f"Failed to retrieve reactors. Status code: {response.status_code}")
+
+def get_reactor_stats(client, reactor):
+    url = f"http://pioreactor.local/api/units/{reactor}/jobs/running"
+    headers = {"Content-Type": "application/json"}
+    
+    response = requests.get(url, headers=headers)
+    if response.status_code == 200:
+        stats = response.json()
+        client.publish('pioreactor/stats', json.dumps(stats))
+    else:
+        print(f"Failed to retrieve reactor stats. Status code: {response.status_code}")
+
+def get_task_status(task_id):
+    # Construct the URL for getting the task status
+    url = f"http://pioreactor.local/unit_api/task_results/{task_id}"
```

Comment:
perhaps
stem="http://pioreactor.local"
at top level
then
url = f"{stem}/.../..."

If that doesn't seem necessary based on your experience with it, feel free to ignore

Reply:
In regards to the global variables I tend to use less of them, kind of a bad practice in general. I feel like have all the info self contained will allow for us to easily move these functions around in the future or if changes needs to be made it is all in one place and easy to see.